### PR TITLE
Bugfix/sof 5578 wip - Make Colab Great Again

### PR DESCRIPTION
**Pull request by timurbazhirov**

### CODE REVIEW

File: requirements-colab.txt
Comment (timurbazhirov) [inline requirements-colab.txt:21]: What i meant earlier is that we don't really need all these packages - instead of relaxing the versions (which I think we should have fixed), we should only have the packages that we actually require. For instance, `ruamel.yaml` and `spglib` and `monty` are all dependencies of pymatgen.

I think, the list should only contain:

- exabyte-api-client
- ase
- pymatgen
- maybe one or two other entries

Comment (timurbazhirov) [reply requirements-colab.txt:21]: Maybe pandas

Comment (timurbazhirov) [reply requirements-colab.txt:21]: +rdkit

File: examples/job/get-file-from-job.py
Comment (timurbazhirov) [inline examples/job/get-file-from-job.py:59]: Let's update the py sources based on ipynb files, pls

File: scripts/env.sh
Comment (timurbazhirov) [inline scripts/env.sh:30]: `IS_USING_GIT_LFS`

File: scripts/env.sh
Comment (timurbazhirov) [inline scripts/env.sh:21]: `IS_RUNNING_ON_COLAB`

File: scripts/env.sh
Comment (timurbazhirov) [inline scripts/env.sh:28]: Let's just put `setup.py` to the root

File: scripts/env.sh
Comment (timurbazhirov) [inline scripts/env.sh:35]:
```
`notebook-info` command is defined in setup.py via `console_scripts`
```
Maybe `notebook-path`

File: examples/job/create_and_submit_job.ipynb
Comment (timurbazhirov) [inline examples/job/create_and_submit_job.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #9.        os.environ.update(
</pre><p>Test Comment</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/71/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

File: scripts/env.sh
Comment (mrakitin) [reply scripts/env.sh]: Addressed via https://github.com/Exabyte-io/api-examples/pull/71/commits/b6eb23abbffceada246a1da9637958b371bd7876.

File: scripts/env.sh
Comment (mrakitin) [reply scripts/env.sh]: Addressed via https://github.com/Exabyte-io/api-examples/pull/71/commits/b6eb23abbffceada246a1da9637958b371bd7876.

File: scripts/env.sh
Comment (mrakitin) [reply scripts/env.sh]: Addressed via https://github.com/Exabyte-io/api-examples/pull/71/commits/b6eb23abbffceada246a1da9637958b371bd7876.

**Comment by mrakitin** [comment]
TODO: before merging, run the following command:
```console
$ scripts/change-branch-in-urls.sh bugfix/SOF-5578-WIP dev
```

File: examples/material/import_materials_from_materialsproject.ipynb
Comment (mrakitin) [reply examples/material/import_materials_from_materialsproject.ipynb:1]: Done via https://github.com/Exabyte-io/api-examples/pull/71/commits/a4259e5ee36373d7fd1063fb9745667e72178771 and https://github.com/Exabyte-io/api-examples/pull/71/commits/45b85e2fa94fe6099ad95d89b13cfa627ce0bae7.

File: scripts/env.sh
Comment (mrakitin) [reply scripts/env.sh]: Done via https://github.com/Exabyte-io/api-examples/pull/71/commits/e9d9cb18328d885730e05f8f39177e84edbaf66c and subsequent commits to resolve minor importing issues.

File: examples/job/get-file-from-job.ipynb
Comment (spectre007) [inline examples/job/get-file-from-job.ipynb:1]: <p>There are still some references to "Exabyte ...", etc. I think we should adjust those to "Mat3ra ..."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/71/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

File: examples/job/ml-train-model-predict-properties.ipynb
Comment (spectre007) [inline examples/job/ml-train-model-predict-properties.ipynb:1]: <p>Exabyte platform -&gt; Mat3ra platform</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/71/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (spectre007) [inline examples/job/ml-train-model-predict-properties.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #6.    from examples.utils.generic import dataframe_to_html, copy_bank_workflow_by_system_name, wait_for_jobs_to_finish, get_property_by_subworkow_and_unit_indicies, display_JSON
</pre><p> Some of these functions seem to be generic and useful enough that we could move them to the API client package. What do you think?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/71/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

File: examples/material/api_interoperability_showcase.ipynb
Comment (spectre007) [inline examples/material/api_interoperability_showcase.ipynb:5]: <pre class="ql-syntax" spellcheck="false">Line #9.    exec(urllib.request.urlopen('https://raw.githubusercontent.com/Exabyte-io/api-examples/dev/examples/utils/initialize_settings.py').read())
</pre><p><code>initialize_settings.py</code> just runs the <code>execute()</code> function and sets an environment variable. Loading a python file and executing its contents does not seem to be the best way. Can we not just import the execute function and run it here? This would require a small adjustment to the API client and is thus perhaps better dealt with in a follow-up task.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/71/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

File: examples/system/get_authentication_params.ipynb
Comment (spectre007) [inline examples/system/get_authentication_params.ipynb:1]: <p>Exabyte account -&gt; Mat3ra account</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/71/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

File: examples/utils/initialize_settings.py
Comment (spectre007) [inline examples/utils/initialize_settings.py:84]: I find this a bit misleading, the function actually returns `None` but prints the path

File: examples/utils/initialize_settings.py
Comment (spectre007) [inline examples/utils/initialize_settings.py:49]: Perhaps we could use type hints here, i.e.
```python
def get_notebook_info() -> dict:
    ...
```

File: examples/utils/initialize_settings.py
Comment (spectre007) [inline examples/utils/initialize_settings.py:76]: If this function doesn't return the path, perhaps we should rename it then to `print_notebook_path()` or so

File: examples/job/create_and_submit_job.ipynb
Comment (mrakitin) [reply examples/job/create_and_submit_job.ipynb:1]: This is a special syntax to make the Google Colab forms.
Docs/examples: https://colab.research.google.com/notebooks/forms.ipynb
Example view:
image

I don't think the original intent was to enable type annotations here.

File: examples/job/ml-train-model-predict-properties.ipynb
Comment (mrakitin) [reply examples/job/ml-train-model-predict-properties.ipynb:1]: Good idea! I support moving generally useful functions to the client package, which will simplify this repo.

File: examples/material/api_interoperability_showcase.ipynb
Comment (mrakitin) [reply examples/material/api_interoperability_showcase.ipynb:5]: This is the old approach which I haven't touched in this file as it was not listed in the table in README. I'll discuss with @timurbazhirov what to do with this example.

The current approach of getting the remote code we agreed on is listed in the first cells of the notebooks from the table, e.g.:
```py
#@title Authorization Form
ACCOUNT_ID = "ACCOUNT_ID" #@param {type:"string"}
AUTH_TOKEN = "AUTH_TOKEN" #@param {type:"string"}
MATERIALS_PROJECT_API_KEY = "MATERIALS_PROJECT_API_KEY" #@param {type:"string"}
ORGANIZATION_ID  = "ORGANIZATION_ID" #@param {type:"string"}

import os
if "COLAB_JUPYTER_IP" in os.environ:
    os.environ.update(
        dict(
            ACCOUNT_ID=ACCOUNT_ID,
            AUTH_TOKEN=AUTH_TOKEN,
            MATERIALS_PROJECT_API_KEY=MATERIALS_PROJECT_API_KEY,
            ORGANIZATION_ID=ORGANIZATION_ID,
        )
    )

    !GIT_BRANCH="bugfix/SOF-5578-WIP"; export GIT_BRANCH; curl -s "https://raw.githubusercontent.com/Exabyte-io/api-examples/${GIT_BRANCH}/scripts/env.sh" | bash
```

File: examples/utils/initialize_settings.py
Comment (mrakitin) [reply examples/utils/initialize_settings.py:84]: You are right, I went back and forth with the function between the `return` and `print`, and forgot to update the docstring. My comment below indicates that the `return` approach results in the non-zero return code when the function is executed via the entry point in `setup.py`, so I switched to the `print` approach. I'll fix the docstring.

File: examples/job/ml-train-model-predict-properties.ipynb
Comment (mrakitin) [reply examples/job/ml-train-model-predict-properties.ipynb:1]: Addressed via 0179594.

File: examples/system/get_authentication_params.ipynb
Comment (mrakitin) [reply examples/system/get_authentication_params.ipynb:1]: Addressed via ee685be.

File: examples/utils/initialize_settings.py
Comment (mrakitin) [reply examples/utils/initialize_settings.py]: Addressed via 6078b14.

File: examples/utils/initialize_settings.py
Comment (mrakitin) [reply examples/utils/initialize_settings.py]: Addressed via 6078b14.

File: examples/utils/initialize_settings.py
Comment (mrakitin) [reply examples/utils/initialize_settings.py]: Addressed via 6078b14.

**Comment by mrakitin** [comment]
> TODO: before merging, run the following command:
> 
> ```
> $ scripts/change-branch-in-urls.sh bugfix/SOF-5578-WIP dev
> ```

Just a reminder for myself and reviewers that the command above needs to be run to fix the branch before merging to `dev`.

File: examples/utils/initialize_settings.py
Comment (chu-k) [inline examples/utils/initialize_settings.py:125]: Minor thing, initialize_settings.py isn't being executed directly anymore in notebooks, we could remove it altogether

**Review by chu-k — APPROVED** [review]
Looks good

File: examples/job/get-file-from-job.py
Comment (timurbazhirov) [inline examples/job/get-file-from-job.py:73]: Should we strip out the numbers as it was in https://github.com/Exabyte-io/api-examples/blob/6ca21439ad2239bf82ade75ac0a34ccfd17bfcb0/examples/config.py#L33 ?

File: examples/utils/initialize_settings.py
Comment (timurbazhirov) [reply examples/utils/initialize_settings.py:125]: @mrakitin - should we remove it?

Comment (mrakitin) [reply examples/utils/initialize_settings.py:125]: Thanks for your review, @chu-k!

I defined a couple of functions that are used in CLI and in some notebooks, but not the old `execute()` one. I can move those to `__init__.py` if people are OK with it, and then the `initialize_settings.py` can be removed.

File: scripts/change-branch-in-urls.sh
Comment (mrakitin) [reply scripts/change-branch-in-urls.sh:40]: @timurbazhirov, looks like the reference is pointing to a different piece of code. Were you referring to https://github.com/Exabyte-io/api-examples/pull/71#discussion_r1235782801?

Comment (timurbazhirov) [reply scripts/change-branch-in-urls.sh:40]: Ok, let's keep as is

File: examples/utils/initialize_settings.py
Comment (timurbazhirov) [reply examples/utils/initialize_settings.py:125]: Let's remove unused functions and rename this file to `notebook.py` maybe?

**Comment by mrakitin** [comment]
All review comments have been addressed, ready for final review, before I update the branch (https://github.com/Exabyte-io/api-examples/pull/71#issuecomment-1597713329).

**Comment by timurbazhirov** [comment]
This can be merged

**Review by spectre007 — APPROVED** [review]
looks good

**Comment by mrakitin** [comment]
Merged, thanks, All!

**Pull request merged**
